### PR TITLE
Refactor jwt #144

### PR DESCRIPTION
**Pull request by KendallWeihe**
# Background

This is a part of a multi-PR (& multi-crate) initiative to complete #143 #144 & #150 without breaking anything, whereafter we'll come through and remove old code. This is the reasoning for `v2` in the file name `lib_v2.rs` -- as to not break anything but to see through the initiative in reviewable PRs. This PR addresses #144. 

# Proposed changes

Notice [the comment here](https://github.com/TBD54566975/web5-rs/issues/144#issuecomment-2098505777) -- after speaking w/ @decentralgabe it's clear we have a requirement to structure the code such that we can add support for JWE at a future date, ideally without inducing breaking changes. I've moved forward with option (3) from that comment, to add future extensibility via namespacing the JWT implementation for JWS's. This is why the `Jwt` implementation is inside of the `jws.rs` file, and why there is an empty `jwe.rs` file; at a future date we can add JWT's as JWE inside of the `jwe.rs` file.

The implementation of JWT's as JWS piggy backs off of the new feature set in the `jws` crate made available from #162.

# Additional work items

Following the work here, we'll implement #150 and subsequently we'll go through and remove all of the old ("non-v2") code.

I've also opened a new ticket to discuss a design change around our rust crate approach #171

### CODE REVIEW

File: crates/jwt/src/lib_v2.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+impl From<SerdeJsonError> for JwtError {
+    fn from(err: SerdeJsonError) -> Self {
+        JwtError::SerdeJsonError(err.to_string())
+    }
+}
+
+pub trait Claims: Serialize + DeserializeOwned {}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct BaseClaims {
```
Comment (KendallWeihe) [inline crates/jwt/src/lib_v2.rs:25]: I want to rename this to `RegisteredClaims` as to match the spec

File: crates/jwt/src/jwe.rs [hunk 2]
```diff
@@ -0,0 +1 @@
+// implementation for JWT's as JWE's
```
Comment (KendallWeihe) [inline crates/jwt/src/jwe.rs:1]: torn on if we should leave this or remove it, I'm leaning towards remove it

File: crates/jwt/src/lib_v2.rs (hunk 1 above)
Comment (KendallWeihe) [reply crates/jwt/src/lib_v2.rs]: done

File: crates/jwt/src/jwe.rs (hunk 2 above)
Comment (KendallWeihe) [reply crates/jwt/src/jwe.rs]: decided to stub in scaffolded code, but not implement and instead created ticket for subsequent work #174

**Comment by nitro-neal** [comment]
this is great, but if the complexity gets too high or there is too much duplicate code it may be better just to do this:

```
One struct with differently named functions Jwt::sign_jws()/Jwt::sign_jwe()/etc (better IMO)
```

In js can do something like this, I don't think rust can do this: Jwt::sign(jweClaim | jwsClaims)

Another approach is have Jwt::sign() which is default a jws (because like 90% of things will use this probabbly)

but have an optional flag or options object that can change it to jwe

Just some thoughts! But I'm fine with the current impl

**Comment by KendallWeihe** [comment]
honestly, I kinda like `Jwt::sign_jws()` and `Jwt::sign_jwe()` better than how it is now with the namespace `jwt::jwe::Jwt` 

opened a ticket for it https://github.com/TBD54566975/web5-rs/issues/177

**Review by nitro-neal — APPROVED** [review]

**Pull request merged**